<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to the `{rmexact}` R Package • rmexact</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to the `{rmexact}` R Package">
<meta property="og:description" content="rmexact">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">rmexact</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/rmexact-intro.html">Introduction to the `{rmexact}` R Package</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to the <code>{rmexact}</code> R
Package</h1>
            
      
      
      <div class="hidden name"><code>rmexact-intro.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This package contains three tests to evaluate the assumption of
random mating in autotetraploids, which are organisms with four copies
of homologous chromosomes. Let <span class="math inline">\(y = (y_0,
y_1, y_2, y_3, y_4)\)</span> be the vector of genotype counts, where
<span class="math inline">\(y_k\)</span> is the number of individuals in
our sample with genotye <span class="math inline">\(k\)</span>. Let
<span class="math inline">\(n = y_0 + y_1 + y_2 + y_3 + y_4\)</span> be
the sample size. Let <span class="math inline">\(q = (q_0, q_1, q_2,
q_3, q_4)\)</span> be the genotype frequencies of the population. Let
<span class="math inline">\(p = (p_0, p_1, p_2)\)</span> be the gamete
frequencies, where <span class="math inline">\(p_k\)</span> is the
proportion of gametes produced in the population with genotype <span class="math inline">\(k\)</span>. This package evaluates the two models
<span class="math display">\[
H_0: y \sim \text{Multinom}(n, p * p)\\
H_A: y \sim \text{Multinom}(n, q)
\]</span> where “<span class="math inline">\(*\)</span>” denotes
discrete linear convolution, and <span class="math inline">\(\text{Multinom}\)</span> represents the
multinomial distribution. This package takes three approaches:</p>
<ol style="list-style-type: decimal">
<li>
<code><a href="../reference/tetexact.html">tetexact()</a></code>: An exact test using sufficient statistics
for tetraploids.</li>
<li>
<code><a href="../reference/rmslrt.html">rmslrt()</a></code>: An exact test using the split likelihood
ratio (Wasserman et al., 2020) for any even ploidy level.</li>
<li>
<code><a href="../reference/rmchisq.html">rmchisq()</a></code>: A standard chi-squared test for any ploidy
level.</li>
</ol>
<p>This vignette will cover these three tests. Let’s load the package
into R</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">rmexact</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="an-exact-test-using-sufficient-statistics">An Exact Test Using Sufficient Statistics<a class="anchor" aria-label="anchor" href="#an-exact-test-using-sufficient-statistics"></a>
</h2>
<p><code><a href="../reference/tetexact.html">tetexact()</a></code>: This function will calculate the exact <span class="math inline">\(p\)</span>-value using sufficient statistics. A
vector containing the genotype counts should be inserted to run this
function. If the user desires to obtain a log <span class="math inline">\(p\)</span>-value, set <code>log_p = TRUE</code> to
make it happen.</p>
<p>Below is an example:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">18</span>, <span class="fl">44</span>, <span class="fl">69</span>, <span class="fl">53</span>, <span class="fl">16</span><span class="op">)</span> </span>
<span><span class="fu"><a href="../reference/tetexact.html">tetexact</a></span><span class="op">(</span>y <span class="op">=</span> <span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.4271683</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="an-exact-test-using-the-split-likelihood-ratio">An Exact Test Using the Split Likelihood Ratio<a class="anchor" aria-label="anchor" href="#an-exact-test-using-the-split-likelihood-ratio"></a>
</h2>
<p><code><a href="../reference/rmslrt.html">rmslrt()</a></code>: This function will calculate the <span class="math inline">\(p\)</span>-value using the split likelihood ratio
test. A vector containing the genotype counts, the proportional split,
and the number of iterations should be inserted to run this function. If
the user desires to obtain a log <span class="math inline">\(p\)</span>-value, set <code>log_p = TRUE</code> to
make it happen.</p>
<p>Below is an example:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nvec1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">18</span>, <span class="fl">44</span>, <span class="fl">69</span>, <span class="fl">53</span>, <span class="fl">16</span><span class="op">)</span> </span>
<span><span class="fu"><a href="../reference/rmslrt.html">rmslrt</a></span><span class="op">(</span>nvec <span class="op">=</span> <span class="va">nvec1</span>, sprop <span class="op">=</span> <span class="fl">0.5</span>, nrep <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="co">#&gt; $q1</span></span>
<span><span class="co">#&gt; [1] 0.08 0.20 0.39 0.26 0.07</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $q2</span></span>
<span><span class="co">#&gt;                                0          1          2 </span></span>
<span><span class="co">#&gt; 0.09115473 0.23619918 0.33833740 0.24011024 0.09419845 </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $nvec1</span></span>
<span><span class="co">#&gt; [1]  8 20 39 26  7</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $nvec2</span></span>
<span><span class="co">#&gt; [1] 10 24 30 27  9</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $pval</span></span>
<span><span class="co">#&gt; [1] 1</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="a-chi-squared-test">A Chi-squared Test<a class="anchor" aria-label="anchor" href="#a-chi-squared-test"></a>
</h2>
<p><code>rmchisqr()</code>: This function will calculate the chi-squared
<span class="math inline">\(p\)</span>-value. A vector containing the
genotype counts should be inserted to run this function. If the user
desires to obtain a log <span class="math inline">\(p\)</span>-value,
set <code>log_p = TRUE</code> to make it happen.</p>
<p>Below is an example:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nvec</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">18</span>, <span class="fl">44</span>, <span class="fl">69</span>, <span class="fl">53</span>, <span class="fl">16</span><span class="op">)</span> </span>
<span><span class="fu"><a href="../reference/rmchisq.html">rmchisq</a></span><span class="op">(</span>nvec <span class="op">=</span> <span class="va">nvec</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 0.6544526</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<ul>
<li>Wasserman, L., Ramdas, A., &amp; Balakrishnan, S. (2020). Universal
inference. <em>Proceedings of the National Academy of Sciences</em>,
117(29), 16880-16890. <a href="doi:10.1073/pnas.1922664117" class="uri">doi:10.1073/pnas.1922664117</a>.</li>
</ul>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by David Gerard, Karene Matoka Nana, NSF DBI 2132247.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
